test/test/test_execution_expectations.rb

```diff
@@ -663,6 +663,43 @@ def return_object
 
                         assert_equal "10", expectation.to_s
                     end
+                    it "validates when the event is emitted" do
+                        plan.add(generator = EventGenerator.new)
+                        expect_execution { generator.emit }
+                            .to { maintain(at_least_until: generator) { true } }
```

Comment:
I would assume the caller would want the predicate to at least match once. Not sure what's your use-case though.

Reply:
Good point. Will fix.